tests/tt_metal/tt_metal/unit_tests_common/basic/test_device_init.cpp

```diff
@@ -0,0 +1,120 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <functional>
+#include <random>
+
+#include "tt_metal/detail/tt_metal.hpp"
+#include "tt_metal/host_api.hpp"
+#include "tt_metal/hostdevcommon/common_runtime_address_map.h"  // FIXME: Should remove dependency on this
+#include "tt_metal/test_utils/env_vars.hpp"
+#include "tt_metal/impl/dispatch/command_queue.hpp"
+#include "tt_metal/test_utils/stimulus.hpp"
+#include "tt_metal/test_utils/print_helpers.hpp"
+
+using namespace tt;
+using namespace tt::test_utils;
+
+namespace unit_tests_common::basic::test_device_init{
+
+void launch_program(tt_metal::Device *device, tt_metal::Program &program) {
+    if (getenv("TT_METAL_SLOW_DISPATCH_MODE")){
+        tt_metal::detail::LaunchProgram(device, program);
+    }else {
+        CommandQueue& cq = device->command_queue();
+        EnqueueProgram(cq, program, false);
+        Finish(cq);
+    }
+}
+
+/// @brief load_blank_kernels into all cores and will launch
+/// @param device
+/// @return
+bool load_all_blank_kernels(tt_metal::Device* device) {
+    bool pass = true;
+    tt_metal::Program program = tt_metal::CreateProgram();
+    CoreCoord compute_grid_size = device->compute_with_storage_grid_size();
+    CoreRange all_cores = CoreRange(
+        CoreCoord{.x = 0, .y = 0},
+        CoreCoord{.x = compute_grid_size.x - 1, .y = compute_grid_size.y -1}
+    );
+    CreateKernel(
+        program, "tt_metal/kernels/dataflow/blank.cpp", all_cores,
+        DataMovementConfig{.processor = DataMovementProcessor::RISCV_1, .noc = NOC::RISCV_1_default});
+
+    CreateKernel(
+        program, "tt_metal/kernels/dataflow/blank.cpp", all_cores,
+        DataMovementConfig{.processor = DataMovementProcessor::RISCV_0, .noc = NOC::RISCV_0_default});
+
+    CreateKernel(program, "tt_metal/kernels/compute/blank.cpp", all_cores, ComputeConfig{});
+
+    unit_tests_common::basic::test_device_init::launch_program(device, program);
+    // tt_metal::detail::LaunchProgram(device, program);
+    return pass;
+}
+}  // namespace gtest_unit_tests::basic
+
+TEST(DeviceInit, SingleDeviceInitializeAndTeardown) {
```

Comment:
Is there some way we can use GTest's parameterization capabilities to avoid writing this test and the test below twice?

Reply:
changed